datafusion/core/src/datasource/file_format/json.rs

```diff
@@ -441,4 +446,94 @@ mod tests {
             .collect::<Vec<_>>();
         assert_eq!(vec!["a: Int64", "b: Float64", "c: Boolean"], fields);
     }
+
+    async fn count_num_partitions(ctx: &SessionContext, query: &str) -> Result<usize> {
```

Comment:
Another potential way to figure out the file groups would be to make the physical plan and then walk it to find the `JsonExec` and its number of partitions